pkg/geo/wkt/lex.go

```diff
@@ -275,43 +313,164 @@ func (l *wktLex) isValidStrideForLayout(stride int) bool {
 		return stride == 4
 	default:
 		// This should never happen.
-		panic("unknown geom.Layout")
+		panic(fmt.Sprintf("unknown geom.Layout %d", curLayout))
+	}
+}
+
+func (l *wktLex) isNonEmptyAllowedForLayout() bool {
+	if l.getCurLayoutMustBeEmpty() {
+		if l.getCurLayout() != geom.XYM {
+			panic("mustBeEmpty is true but layout is not XYM")
+		}
+		l.setIncorrectLayoutError(geom.NoLayout, "XYM layout must use the M variant of the geometry type")
```

Comment:
consider rewording this. maybe "Collection object must use the M variant if the collection is of type XYM."?

Reply:
changed to "the M variant is required for non-empty XYM geometries in GEOMETRYCOLLECTIONs"